server/src/main/scala/za/co/absa/atum/server/api/controller/CheckpointControllerImpl.scala

```diff
@@ -33,12 +34,22 @@ class CheckpointControllerImpl(checkpointService: CheckpointService) extends Che
     )
   }
 
-  override def createCheckpointV2(
-    checkpointDTO: CheckpointDTO
-  ): IO[ErrorResponse, SingleSuccessResponse[CheckpointDTO]] = {
-    mapToSingleSuccessResponse(createCheckpointV1(checkpointDTO))
+  override def postCheckpointV2(
+    partitioningId: Long,
+    checkpointV2DTO: CheckpointV2DTO
+  ): IO[ErrorResponse, (SingleSuccessResponse[CheckpointV2DTO], String)] = {
+    for {
+      response <- mapToSingleSuccessResponse(
```

Comment:
I am not sure I understand. We map only successful calls from Unit to CheckpointV2DTO. Otherwise an ErrorResponse is returned (serviceCall method in BaseController).

Reply:
Resolved in conversation.